src/ConcreteRArray.jl

```diff
@@ -394,6 +394,10 @@ function Base.copyto!(
     return dest
 end
 
+function Base.copyto!(dest::Union{AnyConcreteIFRTArray,AnyConcretePJRTArray}, src::AbstractArray)
+    copyto!(dest, to_rarray(src))
```

Comment:
```suggestion
    copyto!(dest, to_rarray(src, sharding=to_reactant_sharding(dest.sharding)))
```

Not sure if that is exactly right or somethign like that needed

Reply:
cc @avik-pal

Reply:
[see the comment above](https://github.com/EnzymeAD/Reactant.jl/pull/942#discussion_r2001389327)